packages/backend-common/src/reading/FetchUrlReader.ts

```diff
@@ -27,6 +27,41 @@ import {
 import path from 'path';
 import { ReadUrlResponseFactory } from './ReadUrlResponseFactory';
 
+const isInRange = (num: number, [start, end]: [number, number]) => {
+  return num >= start && num <= end;
+};
+
+const parsePortRange = (port: string): [number, number] => {
+  const isRange = port.includes('-');
+  if (isRange) {
+    const range = port
+      .split('-')
+      .map(v => parseInt(v, 10))
+      .filter(Boolean) as [number, number];
+    if (range.length !== 2) throw new Error(`Port range is not valid: ${port}`);
+    const [start, end] = range;
+    if (start <= 0 || end <= 0 || start > end)
+      throw new Error(`Port range is not valid: [${start}, ${end}]`);
+    return range;
+  }
+  const parsedPort = parseInt(port, 10);
+  return [parsedPort, parsedPort];
+};
+
+const parsePortPredicate = (port: string | undefined) => {
+  if (port) {
+    const range = parsePortRange(port);
+    return (url: URL) => {
+      if (url.port) return isInRange(parseInt(url.port, 10), range);
+
+      if (url.protocol === 'http:') return isInRange(80, range);
+      if (url.protocol === 'https:') return isInRange(433, range);
```

Comment:
Shouldn't this range be 443?

Reply:
Yes, it is a typo, I correct immediately